src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryAddress.java

```diff
@@ -116,4 +126,22 @@ static void copy(MemoryAddress src, MemoryAddress dst, long bytes) {
         MemoryAddressImpl.copy((MemoryAddressImpl)src, (MemoryAddressImpl)dst, bytes);
     }
 
+    /**
+     * A native memory address instance modelling the {@code NULL} pointer. This address is backed by a memory segment
+     * which can be neither closed, nor dereferenced.
+     */
+    MemoryAddress NULL = MemorySegmentImpl.NOTHING.baseAddress();
```

Comment:
I'd leave this out for now? This seems very C oriented.

Reply:
OK - My initial thought was that NULL is pretty universal, and that having a singleton NULL might be handy. But I guess there's no real need to use such a value.